src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const stateCopies = [];
+
+  for (const action of actions) {
+    const actionType = action.type;
+
+    switch (actionType) {
+      case 'addProperties':
```

Comment:
Move all properties to the object.

Example of usage:
`case ActionType.AddProperties:`